src/lib/rule2E.ts

```diff
@@ -0,0 +1,187 @@
+import {Context, getDefaultContext} from './context';
+import {computeTextAlternative} from './compute_text_alternative';
+
+// Input types that imply role 'textbox' if list attribute is not present,
+// and imply role 'combobox' if list attribute is present.
+const TEXT_INPUT_TYPES = ['email', 'tel', 'text', 'url', 'search'];
+
+/**
+ * Determines whether a given node has role 'textbox' and,
+ * if so, gets the value of that textbox.
+ * @param node - element whose role is being calculated
+ * @return - textbox value if node is a textbox, null otherwise
+ * (null indicates that node is not a textbox).
+ */
+function getValueIfTextbox(node: HTMLElement): string | null {
+  // Explicit role='textbox' (elements not of type <input>) handled by rule2F
```

Comment:
I think the idea here is: for a node to be accepted by E it already needs to be partOfName, in which case it will also be accepted by F whether it allows name from contents or not. So this should handle any explicit role='textbox' elements whose value would be the text alternative of their subtree.

Do you know of any other kinds of elements that can have role='textbox' but whose value would be something other than the subtree text alternative? If so they should be handled here.

I don't think 'value' is well defined in the spec which makes this somewhat ambiguous.